store/bench_test.go

```diff
@@ -0,0 +1,205 @@
+		benchmarks = append(benchmarks, benchmark{name: name, ratio: ratio})
+	}
+	return benchmarks
+}
+
+func remove(s []string, i int) []string {
+	s[i] = s[len(s)-1]
+	return s[:len(s)-1]
+}
+
+func runLoadStoreV1(b *testing.B, operationsCount uint, ratio [4]uint) {
+	db := dbm.NewMemDB()
+	tree, err := iavl.NewMutableTree(db, cacheSize)
+	require.NoError(b, err)
+	store := storev1.UnsafeNewStore(tree)
+
+	opLimit := map[string]uint{}
+	opLimit["Has"] = operationsCount * ratio[0] / 100
+	opLimit["Get"] = operationsCount * ratio[1] / 100
+	opLimit["Set"] = operationsCount * ratio[2] / 100
+	opLimit["Delete"] = operationsCount * ratio[3] / 100
+
+	opCount := map[string]uint{}
+	keys := []string{"Has", "Get", "Set", "Delete"}
+	for j := 0; j < int(operationsCount); j++ {
+		r, _ := crand.Int(crand.Reader, big.NewInt(int64(len(keys))))
+		k := int(r.Uint64())
+		key := keys[k]
+		switch key {
+		case "Has":
+			store.Has(randBytes(12))
+			opCount["Has"] += 1
+			if opCount["Has"] >= opLimit["Has"] {
+				keys = remove(keys, k)
+			}
+		case "Get":
+			store.Get(randBytes(12))
+			opCount["Get"] += 1
+			if opCount["Get"] >= opLimit["Get"] {
+				keys = remove(keys, k)
+			}
+		case "Set":
+			store.Set(randBytes(12), randBytes(50))
+			opCount["Set"] += 1
+			if opCount["Set"] >= opLimit["Set"] {
+				keys = remove(keys, k)
+			}
+		case "Delete":
+			store.Delete(randBytes(12))
+			opCount["Delete"] += 1
+			if opCount["Delete"] >= opLimit["Delete"] {
+				keys = remove(keys, k)
+			}
+		}
+	}
+
+	id := store.Commit()
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err = storev1.LoadStore(db, id, false, storev1.DefaultIAVLCacheSize)
+		require.NoError(b, err)
+	}
+}
+
+func runLoadStoreV2(b *testing.B, operationsCount uint, ratio [4]uint) {
+	db := memdb.NewDB()
+	store, err := storev2.NewStore(db, storev2.DefaultStoreConfig)
+	require.NoError(b, err)
+
+	opLimit := map[string]uint{}
+	opLimit["Has"] = operationsCount * ratio[0] / 100
+	opLimit["Get"] = operationsCount * ratio[1] / 100
+	opLimit["Set"] = operationsCount * ratio[2] / 100
+	opLimit["Delete"] = operationsCount * ratio[3] / 100
+
+	opCount := map[string]uint{}
+	keys := []string{"Has", "Get", "Set", "Delete"}
+	for j := 0; j < int(operationsCount); j++ {
+		r, _ := crand.Int(crand.Reader, big.NewInt(int64(len(keys))))
+		k := int(r.Uint64())
+		key := keys[k]
+		switch key {
+		case "Has":
+			store.Has(randBytes(12))
+			opCount["Has"] += 1
+			if opCount["Has"] >= opLimit["Has"] {
+				keys = remove(keys, k)
+			}
+		case "Get":
+			store.Get(randBytes(12))
+			opCount["Get"] += 1
+			if opCount["Get"] >= opLimit["Get"] {
+				keys = remove(keys, k)
+			}
+		case "Set":
+			store.Set(randBytes(12), randBytes(50))
+			opCount["Set"] += 1
+			if opCount["Set"] >= opLimit["Set"] {
+				keys = remove(keys, k)
+			}
+		case "Delete":
+			store.Delete(randBytes(12))
+			opCount["Delete"] += 1
+			if opCount["Delete"] >= opLimit["Delete"] {
+				keys = remove(keys, k)
+			}
+		}
+	}
```

Comment:
this code is repeated. Let's define a common interface for `store` and `store_v2` and use it in a single benchmark function.

Reply:
now we have store interface for both storev1 and storev2